main.go

```diff
@@ -74,6 +77,9 @@ func main() {
 		LeaderElection:     enableLeaderElection,
 		LeaderElectionID:   "cc88008e.cluster.x-k8s.io",
 		Namespace:          watchNamespace,
+		ClientDisableCacheFor: []client.Object{
+			&corev1.Secret{},
```

Comment:
It disables caching for secrets which the etcdadm controller relies on to write the etcd endpoints. This is also a precautionary change so the etcd endpoints are always up to date and never pull from out of date cache. Again, I didn't see any significant performance issues here either when creating 10 workload clusters with and without this change.
As for assumptions in the code, I don't think disabling cache for a particular object would affect any existing code paths.

Reply:
ah interesting
What's the scenario where this might be problematic? Cache should not affect if all this controller does is to write. The controller **reading** this secret should be the one being careful with the cache, right?

Reply:
fair point, this controller just writes the secrets so it shouldn't use cache for that. I have removed it